modules/couchbase/src/test/java/org/testcontainers/couchbase/CouchbaseContainerTest.java

```diff
@@ -45,37 +45,60 @@ public void testBasicContainerUsage() {
                 .withBucket(bucketDefinition)
             // }
         ) {
-            container.start();
-
-            // cluster_creation {
-            CouchbaseEnvironment environment = DefaultCouchbaseEnvironment
-                .builder()
-                .bootstrapCarrierDirectPort(container.getBootstrapCarrierDirectPort())
-                .bootstrapHttpDirectPort(container.getBootstrapHttpDirectPort())
-                .build();
-
-            Cluster cluster = CouchbaseCluster.create(
-                environment,
-                container.getHost()
-            );
-            // }
+            setUpClient(container, cluster -> {
+                Bucket bucket = cluster.bucket(bucketDefinition.getName());
+                bucket.waitUntilReady(Duration.ofSeconds(10L));
+
+                Collection collection = bucket.defaultCollection();
+
+                collection.upsert("foo", JsonObject.create().put("key", "value"));
+
+                JsonObject fooObject = collection.get("foo").contentAsObject();
+
+                assertEquals("value", fooObject.getString("key"));
+            });
+        }
+    }
+
+    @Test
+    public void testBucketIsFlushableIfEnabled() {
+        BucketDefinition bucketDefinition = new BucketDefinition("mybucket")
+            .withFlushEnabled(true);
 
-            try {
-                // auth {
-                cluster.authenticate(container.getUsername(), container.getPassword());
-                // }
+        try (
+            CouchbaseContainer container = new CouchbaseContainer(COUCHBASE_IMAGE)
+                .withBucket(bucketDefinition)
+        ) {
+            setUpClient(container, cluster -> {
+                Bucket bucket = cluster.bucket(bucketDefinition.getName());
+                bucket.waitUntilReady(Duration.ofSeconds(10L));
+
+                Collection collection = bucket.defaultCollection();
 
-                Bucket bucket = cluster.openBucket(bucketDefinition.getName());
+                collection.upsert("foo", JsonObject.create().put("key", "value"));
 
-                bucket.upsert(JsonDocument.create("foo", JsonObject.empty()));
+                cluster.buckets().flushBucket(bucketDefinition.getName());
```

Comment:
this is ok in theory, but flushBucket is async on the server so performing a set afterwards immediately might fail. Could use something like awaitility to execute the exists for some time in a loop until the condition is matched to be a little more resilient to races.

Reply:
Done :)